### PR TITLE
better api for mermaid support

### PR DESCRIPTION
**Pull request by colombod**

### CODE REVIEW

File: src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/MermaidKernelExtensionTests.cs [hunk 1]
```diff
@@ -123,7 +172,24 @@ public void can_generate_class_diagram_from_generic_type()
         [Fact]
         public void can_generate_class_diagram_to_a_specified_depth()
         {
-            var diagram = typeof(List<Dictionary<string, object>>).ToClassDiagram(graphDepth:2);
+            var diagram = typeof(List<Dictionary<string, object>>).ToClassDiagram(new ClassDiagramConfiguration(2));
+
+            this.Assent(diagram.ToString());
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/MermaidKernelExtensionTests.cs:177]: I would do this without `Assent`. Maybe verify that a specific 3rd-level item isn't present?

Comment (colombod) [reply src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/MermaidKernelExtensionTests.cs:177]: that means parsing lines :(

Comment (colombod) [reply src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/MermaidKernelExtensionTests.cs]: I made it

File: src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/MermaidKernelExtensionTests.cs [hunk 2]
```diff
@@ -70,6 +70,23 @@ graph TD
...
+                new CSharpKernel().UseNugetDirective(),
+            };
+
+            var extension = new MermaidKernelExtension();
+
+            await extension.OnLoadAsync(kernel);
+
+            string formatted = typeof(List<string>).ExploreWithUmlClassDiagram().ToDisplayString(HtmlFormatter.MimeType);
+
+            this.Assent(formatted.FixedGuid());
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/MermaidKernelExtensionTests.cs:87]: This test also doesn't really verify what the name of the test is verifying. It's verifying that the output matches the approval, which isn't readable from here. It's not verifying the content. For example, any object can be formatted to HTML using our default formatters. We should probably verify something about the content itself.

Comment (colombod) [reply src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/MermaidKernelExtensionTests.cs:87]: Ok will fix it later

Comment (colombod) [reply src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/MermaidKernelExtensionTests.cs]: changed

**Review by jonsequitur — APPROVED** [review]

**Pull request merged**